flake.nix

```diff
@@ -0,0 +1,50 @@
+{
+  description = "The baz editor";
+
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.flake-utils.follows = "flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = {
+    self,
+    flake-utils,
+    nixpkgs,
+    rust-overlay,
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [
+          (import rust-overlay)
+        ];
+      };
+
+      rustToolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+    in {
+      devShells.default = pkgs.mkShell {
+        buildInputs = with pkgs; [
+          glib
+          gtk3
+          libsoup
+          nodejs
```

Comment:
missing `nodePackages.npm` and `nodePackages.svelte-language-server` and I believe rust-analyzer